pageserver/src/tenant/layer_map_mgr.rs

```diff
@@ -141,6 +237,33 @@ mod tests {
                 .key_range,
             Key::from_i128(1)..Key::from_i128(2)
         );
+
+        assert_eq!(mgr.lowest_version_in_use(), 1);
+        drop(ref_1);
+        drop(ref_2);
+        assert_eq!(mgr.lowest_version_in_use(), 2);
+        mgr.update(|map| async move { Ok(map) }).await?;
+        assert_eq!(mgr.lowest_version_in_use(), 3);
+
         Ok(())
     }
+
+    #[test]
+    fn test_watermark() {
+        let watermark = LayerMgrWatermark::new(0);
+        assert_eq!(watermark.lowest_version_in_use(), 0);
+        watermark.release(0);
+        assert_eq!(watermark.lowest_version_in_use(), 1);
+        watermark.release(1);
+        assert_eq!(watermark.lowest_version_in_use(), 2);
+        watermark.release(3);
+        watermark.release(4);
```

Comment:
```suggestion
        watermark.release(4);
        watermark.release(4);
        watermark.release(4);
```

Let's see what happens when we have multiple layer map guards at the same version and we drop all of them